File: VB/HierarchicalDataAdapterSample/DataObjects.vb
Imports System.Collections.Generic

Namespace HierarchicalDataAdapterSample
    Public Class EnergyInfo
        Public Property Type() As String
        Public Property Value() As Double
    End Class

    Public Class CountryStatistics
        Public Property Name() As String


        Private energyStatistics_Renamed As New List(Of EnergyInfo)()
        Public ReadOnly Property EnergyStatistics() As List(Of EnergyInfo)
            Get
                Return energyStatistics_Renamed
            End Get
        End Property
    End Class
End Namespace

File: CS/HierarchicalDataAdapterSample/DataObjects.cs
using System.Collections.Generic;

namespace HierarchicalDataAdapterSample {
    public class EnergyInfo {
        public string Type { get; set; }
        public double Value { get; set; }
    }

    public class CountryStatistics {
        public string Name { get; set; }

        List<EnergyInfo> energyStatistics = new List<EnergyInfo>();
        public List<EnergyInfo> EnergyStatistics { get { return energyStatistics; } }
    }
}
